sdk/src/instruction.rs

```diff
@@ -86,18 +86,16 @@ pub enum InstructionError {
     /// the runtime cannot determine which changes to pick or how to merge them if both are modified
     DuplicateAccountOutOfSync,
 
-    /// CustomError allows on-chain programs to implement program-specific error types and see
-    /// them returned by the Solana runtime. A CustomError may be any type that is represented
-    /// as or serialized to a u32 integer.
-    ///
-    /// NOTE: u64 requires special serialization to avoid the loss of precision in JS clients and
-    /// so is not used for now.
+    /// Allows on-chain programs to implement program-specific error types and see them returned
+    /// by the Solana runtime. A program-specific error may be any type that is represented as
+    /// or serialized to a u32 integer.
     CustomError(u32),
 
-    /// Like CustomError but the return value from the program conflicted with
-    /// a builtin error.  The value held by this variant is the u32 error code
-    /// returned by the program but with the 30th bit cleared.
-    ConflictingError(u32),
+    /// Like `CustomError` but the return value from the program was invalid.  Valid
+    /// errors are either a defined builtin error value or a user-defined error in
+    /// the lower 32 bits.  The value held by this variant is the upper 32 bits of the value
+    /// returned by the program.
+    InvalidError(u32),
```

Comment:
Should we return the full error code here? 

```rust
InvalidError { upper: u32, lower: u32 },
```

Could be abused I suppose. We could just call it OverflowError or BigError too 🤔

Reply:
P.S. If we do return the full error code, we should add back the javascript u64 note

Reply:
Depends on how big we want to bloat error, also thinking about adding another layer of `ProgramError` too...